NotesApi.Tests/V1/Controllers/NotesApiControllerTests.cs

```diff
@@ -85,5 +89,19 @@ public void GetPersonByIdAsyncExceptionIsThrown(string paginationToken)
             // Assert
             func.Should().Throw<ApplicationException>().WithMessage(exception.Message);
         }
+
+        [Fact]
+        public void PostNewNoteAsyncExceptionIsThrown()
+        {
+            // Arrange
+            var exception = new ApplicationException("Test exception");
+            _mockPostNewNoteUseCase.Setup(x => x.ExecuteAsync(It.IsAny<CreateNoteRequest>())).ThrowsAsync(exception);
+
+            // Act
+            Func<Task<IActionResult>> func = async () => await _sut.PostNewNote(new CreateNoteRequest()).ConfigureAwait(false);
+
+            // Assert
+            func.Should().Throw<ApplicationException>().WithMessage(exception.Message);
+        }
```

Comment:
We should add another test for successful creation of a note as well

Reply:
Already cover in E2E

Reply:
The E2E tests cover tests to ensure all the different parts work as a whole however we need a unit test to ensure that only the controller works successfully as well, Does that make sense?